src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -57,8 +60,47 @@ public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, stri
 
             _frameworkIdentifier = frameworkIdentifier;
             _frameworkVersion = NormalizeVersion(frameworkVersion);
-            _frameworkProfile = frameworkProfile ?? string.Empty;
+
+            IsNet5Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = profile ?? string.Empty;
+            Platform = string.Empty;
+            PlatformVersion = FrameworkConstants.EmptyVersion;
+        }
+
+        /// <summary>
+        /// Creates a new NuGetFramework instance, with an optional platform and platformVersion (only available for net5.0+)
+        /// </summary>
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string platform, Version platformVersion)
+        {
+            if (frameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("frameworkIdentifier");
```

Comment:
style: Consider using nameof(frameworkIdentifier)

Reply:
I don't think we should change this because the public API has been hard-coded to this string and it might be weird for an overload to have different behavior.

Reply:
Renaming a parameter is technially a breaking change, so we shouldn't do that. 

Not sure how I feel about that motivation in particular. 

Throwing an argument null exception for a parameter name that's potentially wrong is also bad.

Reply:
I tried to do this and... uhh?

![image](https://user-images.githubusercontent.com/17535/80138345-86e59d80-8559-11ea-9ebc-a308da3d682d.png)

Reply:
Waaatt...NuGetFramework is still targetting net40 :D 
Not super relevant in the grand scheme of things. Let's just move on :)